### PR TITLE
Stats: Remove statsAsyncLoading feature flag

### PR DESCRIPTION
**Pull request by ScoutHarris**
Ref #11852 

To test:
- Go to Stats > Insights.
  - Verify ghost views appear when loading.
- On an Insight card, select 'View more'.
  - Verify ghost views appear when loading.

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Stats/Insights/SiteStatsInsightsTableViewController.swift [hunk 1]
```diff
@@ -218,25 +206,13 @@ private extension SiteStatsInsightsTableViewController {
...
-                displayFailureViewIfNecessary()
-            } else {
-                hideNoResults()
-            }
+        if viewModel.fetchingFailed() {
+            displayFailureViewIfNecessary()
+        }
+        else {
+            hideNoResults()
         }
```
Comment (danielebogo) [inline WordPress/Classes/ViewRelated/Stats/Insights/SiteStatsInsightsTableViewController.swift]: I think this should be just
```
if viewModel.fetchingFailed() {
    displayFailureViewIfNecessary()
}
```

File: WordPress/Classes/ViewRelated/Stats/Insights/SiteStatsInsightsTableViewController.swift [hunk 2]
```diff
@@ -597,56 +573,25 @@ extension SiteStatsInsightsTableViewController: NoResultsViewControllerDelegate
...
-        if asyncLoadingActivated {
-            hideNoResults()
-            return
-        }
-
-        updateNoResults(title: NoResultConstants.loadingTitle,
-                        accessoryView: NoResultsViewController.loadingAccessoryView()) { noResults in
-                            noResults.hideImageView(false)
-        }
+        hideNoResults()
```
Comment (danielebogo) [inline WordPress/Classes/ViewRelated/Stats/Insights/SiteStatsInsightsTableViewController.swift]: We can remove the `defer` above and add `addViewModelListeners()` and `refreshInsights()` after `hideNoResults()`

**Comment by ScoutHarris** [comment]
Thanks @danielebogo ! Ready for another review please.

**Comment by danielebogo** [comment]
:shipit:

**Pull request merged**